src/DynamoCore/Core/Threading/UpdateRenderPackageAsyncTask.cs

```diff
@@ -255,28 +259,21 @@ private void GetRenderPackagesFromMirrorData(MirrorData mirrorData, ref List<str
                     // color of 0,0,0,255 (Black), we adjust the color components here.
                     if (graphicItem is Curve || graphicItem is Surface || graphicItem is Solid || graphicItem is Point)
                     {
-                        for (var i = 0; i < package.LineStripVertexColors.Count; i += 4)
-                        {
-                            package.LineStripVertexColors[i] = defR;
-                            package.LineStripVertexColors[i + 1] = defG;
-                            package.LineStripVertexColors[i + 2] = defB;
-                        }
-
-                        for (var i = 0; i < package.PointVertexColors.Count; i += 4)
-                        {
-                            package.LineStripVertexColors[i] = defR;
-                            package.LineStripVertexColors[i + 1] = defG;
-                            package.LineStripVertexColors[i + 2] = defB;
-                        }
+                        if(package.LineVertexCount > 0)
+                            package.ApplyLineVertexColors(0, package.LineVertexCount-1, defR, defG, defB, defA);
+                        if(package.PointVertexCount > 0)
+                            package.ApplyPointVertexColors(0, package.PointVertexCount-1, defR, defG, defB, defA);
```

Comment:
This is more a matter of style, but I find that enclosing an if block in curly brackets makes code both easier to read and less prone to refactoring errors.